code/modules/mining/lavaland/ash_tree.dm

```diff
@@ -0,0 +1,82 @@
+//Houses the ash tree, a lava land tree that has been burning for quite some time making a maple like sweetener.
+
+/obj/structure/flora/ashtree
+	gender = PLURAL //same as other tree
+	icon = 'icons/obj/lavaland/ash_tree.dmi'
+	icon_state = "ashtree"
+	var/sap_icon_state = "ashtree_maple" //Are icon when we are full of honey or other sap
+	var/tabbed_icon_state = "ashtree_maple" //What we look like when tapping
+	name = "ashed tree"
+	desc = "A once large tree now burnt like the lands around it."
+	density = TRUE
+	pixel_x = -16
+	layer = FLY_LAYER
+	var/coal_amount = 5 //amout of coal in are tree, simular to logs
+	var/sap = FALSE //Do we have sap?
+	var/are_sap = /datum/reagent/consumable/honey //What reagent we have
+	//var/tapping_items = list(/obj/item/reagent_containers/glass) - current dosnt work commiting it out
+	var/harvest_sap_time = 60 //This is in seconds, and now long we wait till are tree is tapped
+	var/container_used
+	var/sap_amount
+
+/obj/structure/flora/ashtree/New()
+	..()
+	if(prob(50))
+		sap = TRUE
+		icon_state = sap_icon_state
+		desc = "A once large tree now burnt like the lands around it. This one seems to have a sap still inside."
+		sap_amount = rand(5,15) //good amout of honey
+	coal_amount = rand(5,15) //We give a random amout
+	SSblackbox.record_feedback("tally", "Honey Tree", 1, "Trees Spawned") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+
+//So we dont lose are bowls, stolen form closet code
+/obj/structure/flora/ashtree/Destroy()
+	dump_contents(override = FALSE)
+	return ..()
+
+/obj/structure/flora/ashtree/proc/dump_contents(override = TRUE) //Override is for not revealing the locker electronics when you open the locker, for example
```

Comment:
What is this for?

Reply:
When cutting down a tree, so we dont null are bowl if one is present, of other item inside it